pkg/volume/azure_file/azure_file.go

```diff
@@ -207,7 +209,12 @@ func (b *azureFileMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 	os.MkdirAll(dir, 0750)
-	source := fmt.Sprintf("//%s.file.core.windows.net/%s", accountName, b.shareName)
+
+	azure, err := getAzureCloud(b.plugin.host.GetCloudProvider())
```

Comment:
Consider adding a separate azure cloud provider name parameter instead of using current cluster's.

Azure File is a pervasive service as it uses standard SMB protocol.
Windows client already supports mounting remotely and linux is on the way according to 
[doc](https://docs.microsoft.com/en-us/azure/storage/storage-how-to-use-files-linux).

If limited to current cloud provider, following will break:
1. non-azure backended cluster won't be able to mount Azure File
2. Clusters running on Regional Azure won't be able to mount Azure File Share on global Azure